tools/mdm/migration/mdmproxy/mdmproxy.go

```diff
@@ -0,0 +1,312 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"howett.net/plist"
+)
+
+type mdmProxy struct {
+	migrateUDIDs      map[string]struct{}
+	migratePercentage int
+	existingServerURL string
+	existingHostname  string
+	fleetServerURL    string
+	existingProxy     *httputil.ReverseProxy
+	fleetProxy        *httputil.ReverseProxy
+	// mutex is used to sync reads/updates to the migrateUDIDs and migratePercentage
+	mutex sync.RWMutex
+	// token is used to authenticate updates to the migrateUDIDs and migratePercentage
+	token string
+}
+
+func (m *mdmProxy) handleProxy(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Host != "" {
+		log.Printf("%s %s Forbidden", r.Method, r.URL.String())
+		http.Error(w, "Forbidden", http.StatusForbidden)
+		return
+	}
+
+	// Send all SCEP requests to the existing server
+	if strings.Contains(r.URL.Path, "scep") {
+		log.Printf("%s %s -> Existing (SCEP)", r.Method, r.URL.String())
+		m.existingProxy.ServeHTTP(w, r)
+		return
+	}
```

Comment:
a completely separate discussion, but just so it's in your radar, Ideally all SCEP requests go through Fleet, I think we'll soon know more as part of https://github.com/fleetdm/fleet/issues/19800

Reply:
This will be an easy change, I just didn't know how to test it so left it as-is for now.